assessments/projects/in-toto/self-assesment.md

```diff
@@ -0,0 +1,783 @@
+in-toto metadata using a very simple POST endpoint and filesystem namespacing
+scheme.
+
+#### Security implications of choosing a metadata store
+
+in-toto was designed under the assumption that an attacker may affect the
+storage or transmission of metadata. As such, the security requirements for a
+metadata store are (rather lax). The main features that should be considered
+are replay protection and mix-and-match attacks.
+
+A complete, very slightly outdated supply chain could be replayed by an
+attacker if there is no stronger replay protection, but this is bounded within
+the expiration date. Stronger replay protection can be added on the metadata
+store (e.g., TUF), to mitigate the concern.
+
+Mix-and-match attacks can be performed by selectively replaying link metadata,
+and will be caught by the artifact hash-chaining. Thus, mix-and-match attacks
+are not a concern.
+
+
+## Secure development practices
+
+in-toto’s implementations are aimed to reduce their attack surface in the face
+of untrusted input. Also, the development process is quite documented and
+clear, in order to avoid software supply chain compromises on in-toto itself.
+As such, close attention to its dependencies and their development is crucial.
+
+
+In this section, we will explore the development ecosystem of the in-toto
+project itself, as well as its subprojects and dependencies.
+
+
+### Development Pipeline: main implementation
+
+The main implementation is the one that’s subjected to the most stringent
+security policies. In developing the in-toto Python implementation, we follow
+these practices:
+
+* all code is pushed into a feature branch and reviewed by at least one core
+  member. Pushes to master (develop) are forbidden
+* All pull-requests are run into travis-ci (linux, python 2.7, 3.4, 3.5, 3.6)
+  and appveyor (windows) and more than 240+ unit tests are run
+* Static analysis is performed using python-bandit
+* Code style is performed using pylint following the lab’s canonical pylintrc
+  file
+* A coverage of 100% is required in order to merge any new feature
+* When releasing new versions of the code:
+   * a CHANGELOG.md file is created noting the new features included in this
+     new version
+   * A version bump commit is added
+   * A signed tag is created
+   * .zip and .tar.gz releases are generated and signed with the gpg  keys:
+     `"903BAB73640EB6D65533EFF3468F122CE8162295"`
+     `"8BA69B87D43BE294F23E812089A2AD3C07D962E8"`
+* The resulting releases and signatures are both uploaded to Github and PyPI,
+  as well as other downstream distribution channels such as the Arch Linux
+  repositories.
+
+
+### Communication Channels
+
+#### Internal
+We have a semi-private (publicly-writable) mailing list as a google group at
+in-toto-dev@googlegroups.com
+
+#### Inbound
+
+It is possible to reach out the in-toto team members at:
+
+* #in-toto@irc.freenode.net
+* The #in-toto channel at secure-systems-lab.slack.com

```

Comment:
would be nice to link "secure-systems-lab.slack.com" to wherever it's listed how to get an invite.  If I just go there in slack, it says:
![image](https://user-images.githubusercontent.com/41906/59451319-08a85e80-8dc0-11e9-80c0-11a94c8bff61.png)

Reply:
unless that is intended for core team only and public should access via in-toto freenode, if that's the case, just note that please